packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -395,6 +395,7 @@ class ThemeData {
   }) {
     return new ThemeData(
       brightness: brightness ?? this.brightness,
+      primarySwatch: primarySwatch,
```

Comment:
Aha! As we've just seen a ThemeData object cannot have a null property and we assert as much. So primarySwatch, which is only used to provide default values for unspecified ThemeData constructor parameters, shouldn't be a parameter to ThemeData.copyWith() at all.